recipes/_compute_base_config.rb

```diff
@@ -65,6 +50,33 @@
   shell '/bin/bash'
 end
 
+# Parse shared directory info and turn into an array
+_shared_dir_array = node['cfncluster']['cfn_shared_dir'].split(',')
+_shared_dir_array.each_with_index do |dir, index|
```

Comment:
I'm very new to Ruby but in general I would avoid modifying a collection while iterating over it. 
What about using map or map! (for in place changes)?